packages/aws-cdk/lib/cli/telemetry/endpoint-sink.ts

```diff
@@ -0,0 +1,141 @@
+import type { IncomingMessage } from 'http';
+import type { Agent } from 'https';
+import { request } from 'https';
+import type { UrlWithStringQuery } from 'url';
+import { ToolkitError } from '@aws-cdk/toolkit-lib';
+import { IoHelper } from '../../api-private';
+import type { IIoHost } from '../io-host';
+import type { TelemetrySchema } from './schema';
+import type { ITelemetrySink } from './sink-interface';
+
+const REQUEST_ATTEMPT_TIMEOUT_MS = 500;
+
+/**
+ * Properties for the Endpoint Telemetry Client
+ */
+export interface EndpointTelemetrySinkProps {
+  /**
+   * The external endpoint to hit
+   */
+  readonly endpoint: UrlWithStringQuery;
+
+  /**
+   * Where messages are going to be sent
+   */
+  readonly ioHost: IIoHost;
+
+  /**
+   * The agent responsible for making the network requests.
+   *
+   * Use this to set up a proxy connection.
+   *
+   * @default - Uses the shared global node agent
+   */
+  readonly agent?: Agent;
+}
+
+/**
+ * The telemetry client that hits an external endpoint.
+ */
+export class EndpointTelemetrySink implements ITelemetrySink {
+  private events: TelemetrySchema[] = [];
+  private endpoint: UrlWithStringQuery;
```

Comment:
Should accept a string and `parse` internally. It would avoid exposing nodejs types in the props interface. Also - external callers are not the ones who want to parse, its this class that needs it, so it should do it.